File: src/Samples/XAML_Features/Validation/Person.fs
namespace OpenSilver.Samples.Showcase

open System
open System.ComponentModel

type Person() =
    let mutable name = ""
    let mutable age = 1

    let propertyChanged = Event<PropertyChangedEventHandler, PropertyChangedEventArgs>()
    interface INotifyPropertyChanged with
        [<CLIEvent>]
        member this.PropertyChanged = propertyChanged.Publish

    member this.Name
        with get() = name
        and set value =
            if String.IsNullOrWhiteSpace(value) then
                raise (System.Exception("Name cannot be empty."))
            name <- value
            propertyChanged.Trigger(this, PropertyChangedEventArgs("Name"))

    member this.Age
        with get() = age
        and set value =
            if value <= 0 then
                raise (System.Exception("Age cannot be lower than 0."))
            age <- value
            propertyChanged.Trigger(this, PropertyChangedEventArgs("Age"))

File: src/Samples/XAML_Features/Validation/Person.vb
Imports System.ComponentModel

Namespace Global.OpenSilver.Samples.Showcase
    'Validation:
    Public Class Person
        Implements INotifyPropertyChanged
        Private _name As String
        Public Property Name As String
            Get
                Return _name
            End Get
            Set(ByVal value As String)
                If String.IsNullOrWhiteSpace(value) Then
                    Throw New Exception("Name cannot be empty.")
                End If
                _name = value
                RaisePropertyChanged("Name")
            End Set
        End Property

        Private _age As Integer = 1
        Public Property Age As Integer
            Get
                Return _age
            End Get
            Set(ByVal value As Integer)
                If value <= 0 Then
                    Throw New Exception("Age cannot be lower than 0.")
                End If
                _age = value
                RaisePropertyChanged("Age")
            End Set
        End Property

        Private Sub RaisePropertyChanged(ByVal propertyName As String)
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(propertyName))
        End Sub
        Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged
    End Class
End Namespace
